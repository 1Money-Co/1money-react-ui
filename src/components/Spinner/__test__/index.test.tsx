import 'jsdom-global/register';
import * as React from 'react';
import { configure, shallow, render, mount } from 'enzyme';
import Adapter from '@cfaester/enzyme-adapter-react-18';
import sinon from 'sinon';
import { Spinner } from '../index';

configure({ adapter: new Adapter() });

const originalConsoleError = console.error;
console.error = (message, ...optionalParams) => {
  if (
    message.includes('Could not parse CSS stylesheet') ||
    message.includes('findDOMNode is deprecated and will be removed')
  ) {
      return;
  }
  originalConsoleError(message, ...optionalParams);
};

describe('Spinner', () => {
  it('renders correctly', () => {
    const wrapper = render(
      <Spinner />
    );
    expect(wrapper).toMatchSnapshot();
  });
  it('simulate events', () => {
    const onClick = sinon.spy(); 
    const wrapper = mount(
      <Spinner onClick={onClick} />
    );
    wrapper.find('div').simulate('click');
    expect(onClick.called).toBe(true);
  });
});
