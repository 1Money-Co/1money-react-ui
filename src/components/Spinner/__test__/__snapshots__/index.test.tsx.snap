// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Spinner renders correctly 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        aria-busy="true"
        class="om-react-ui-spinner p-progress-spinner"
        data-pc-name="progressspinner"
        data-pc-section="root"
        role="progressbar"
      >
        <svg
          class="p-progress-spinner-svg"
          data-pc-section="spinner"
          style="animation-duration: 2s;"
          viewBox="25 25 50 50"
        >
          <circle
            class="p-progress-spinner-circle"
            cx="50"
            cy="50"
            data-pc-section="circle"
            fill="none"
            r="20"
            stroke-miterlimit="10"
            stroke-width="2"
          />
        </svg>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      aria-busy="true"
      class="om-react-ui-spinner p-progress-spinner"
      data-pc-name="progressspinner"
      data-pc-section="root"
      role="progressbar"
    >
      <svg
        class="p-progress-spinner-svg"
        data-pc-section="spinner"
        style="animation-duration: 2s;"
        viewBox="25 25 50 50"
      >
        <circle
          class="p-progress-spinner-circle"
          cx="50"
          cy="50"
          data-pc-section="circle"
          fill="none"
          r="20"
          stroke-miterlimit="10"
          stroke-width="2"
        />
      </svg>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
